src/useAsyncFn.ts

```diff
@@ -18,17 +18,33 @@ export type AsyncState<T> =
       value: T;
```

Comment:
Couldn't we just do `value: T | undefined;` on this line so we don't need to define an `InitialState` for the `AsyncFn` type and don't need to defined 2 version of `AsyncState` for the `initialState` parameter? A Promise can resolve `undefined` anyways 🤷‍♂️

Reply:
I agree that Promise can resolve to `undefined`, but in that case `undefined` will be _inside_ `T`.

If we allow `value: T | undefined`, then developers will need to check against `undefined`, i.e. an problem similar to what this issue is about:

![](https://user-images.githubusercontent.com/3994389/64019571-c4f80200-cb2f-11e9-8591-1ae3898c3855.png)